### PR TITLE
Site chat

### PR DESCRIPTION
**Pull request by andygray**
* [x] Chat at entity level
* [x] avatars
* [x] `Users` and `Organisations` factory classes
* [x] fixed icons
* [x] small bug fixes
* [x] organisation detail on site page
* [x] Ready for Review and Merge

### CODE REVIEW

File: app/js/dashboard/chatter-box/ChatterBoxCtrl.js [hunk 1]
```diff
@@ -19,13 +17,28 @@
...
+            // limit: 25, endAt: null, startAt: null, limitToFirst: 25, limitToLast: 25
+            // TODO - limit not correct and limit to first does not seem to work.
+            var fbChatUrl = 'chatter-box';
+            if (entity) {
+                fbChatUrl = fbChatUrl + '/' + entity.$id;
+            }
+            else {
+                fbChatUrl = fbChatUrl + '/dashboard';
+            }
+            vm.chatItems = fbutil.syncArray(fbChatUrl, {orderBy: '$priority'});
```
Comment (jamesmorgan) [inline app/js/dashboard/chatter-box/ChatterBoxCtrl.js]: This should in in a factory or service class I think. Good/easy to user implementation though :+1:

File: app/js/entity/organisation/Organisations.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        var orgs = fbutil.syncArray('organisations');
+
+        return {
+            all: function () {
+                return orgs;
+            },
+            get: function (orgId) {
+                // Simple index lookup
+                 return orgs.$getRecord(orgId);
+            }
```
Comment (jamesmorgan) [inline app/js/entity/organisation/Organisations.js]: I like this pattern, I wonder if I could simplify other services to factories of this style? 

Are these now 3 way bind on the view?

File: app/js/navbar/ProfileCtrl.js [hunk 3]
```diff
@@ -4,12 +4,13 @@
     /**
      * @ngInject
      */
-    function ProfileCtrl(simpleLogin, $state, EditUserModalService) {
+    function ProfileCtrl(simpleLogin, $state, EditUserModalService, UserService) {
```
Comment (jamesmorgan) [inline app/js/navbar/ProfileCtrl.js]: Is `simpleLogin` not used and can be ditched?

File: app/js/routes.js [hunk 4]
```diff
@@ -188,14 +188,10 @@
...
-                        templateUrl: 'partials/site-detail/site-view.html',
-                        controller: 'SiteOverviewCtrl',
-                        controllerAs: 'SiteOverviewCtrl'
+                        templateUrl: 'partials/site-detail/site-view.html'
                     },
                     'organisation-simple-view@site': {
-                        templateUrl: 'partials/organisation-detail/organisation-simple-view.html',
-                        controller: 'SiteOverviewCtrl',
-                        controllerAs: 'SiteOverviewCtrl'
+                        templateUrl: 'partials/organisation-detail/organisation-simple-view.html'
```
Comment (jamesmorgan) [inline app/js/routes.js]: How does the controller get referenced?

File: app/index.html [hunk 5]
```diff
@@ -78,7 +77,7 @@
             <!-- Sidebar user panel -->
             <div class="user-panel">
                 <div class="pull-left image">
-                    <img src="img/avatar3.png" class="img-circle" alt="User Image"/>
+                    <img class="img-circle" src="data:image/jpeg;base64, {{ sc.userData.avatar }}" alt="{{ sc.userData.name }}" title="{{ sc.userData.name }}"/>
```
Comment (jamesmorgan) [inline app/index.html]: How did you get the `avatar` in firebase?

File: app/js/entity/organisation/Organisations.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+     */
+    var Organisations = function ($log, fbutil) {
+
+        var orgs = fbutil.syncArray('organisations');
+
+        return {
+            all: function () {
+                return orgs;
+            },
+            get: function (orgId) {
```
Comment (jamesmorgan) [inline app/js/entity/organisation/Organisations.js]: Is this any different than `OrganisationService.getById()` && `OrganisationService.list()`?

File: app/js/entity/organisation/Organisations.js (hunk 2 above)
Comment (andygray) [inline app/js/entity/organisation/Organisations.js]: Yeah - think so. Will test. But I uploaded an avatar and it appeared before a refresh so guess so...

File: app/js/navbar/ProfileCtrl.js (hunk 3 above)
Comment (andygray) [inline app/js/navbar/ProfileCtrl.js]: yep. Will ditch

Comment (andygray) [inline app/js/navbar/ProfileCtrl.js]: needed for logout - should move these functions as discussed.

File: app/js/routes.js (hunk 4 above)
Comment (andygray) [inline app/js/routes.js]: well, it's from the parent/top-level. I will re-think/discuss this with you.

File: app/index.html (hunk 5 above)
Comment (andygray) [inline app/index.html]: added manually and used online base-64 conversion.

Comment (jamesmorgan) [inline app/index.html]: :+1:

File: app/js/entity/organisation/Organisations.js (hunk 6 above)
Comment (andygray) [inline app/js/entity/organisation/Organisations.js]: Nope. Not really. Maybe not required. Was thinking maybe have these factory classes and keep the service for manipulation i.e. updates/inserts/relationship and remove the loading of the array?

Comment (jamesmorgan) [inline app/js/entity/organisation/Organisations.js]: Not a bad idea will play about on my PR.

**Pull request merged**